### PR TITLE
Enable Support for more than 1 IS31FL3733

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Enable a second IS31FL3733 in rgb_matrix as it is mostly already supported elsewhere in the code base.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Enable second IS31FL3733

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
Also, this should probably target `develop`.

**Comment by Xelus22** [comment]
> 
> 
> The DZtech boards have only one 3733 and `DRIVER_ADDR_2` is set to the same value as `DRIVER_LED_1`, I assume because it doesn't compile without it being defined, even if it isn't used. I wonder if that breaks anything, and whether we should add checks to only run stuff for driver 2 when `DRIVER_COUNT == 2`.

I would assume we would need checks for all our supported drivers then. i.e. 3731, 3733, 3736, 3737 and 3741

File: quantum/rgb_matrix_drivers.c [hunk 1]
```diff
@@ -34,6 +34,7 @@ static void init(void) {
     IS31FL3731_init(DRIVER_ADDR_2);
 #    elif defined(IS31FL3733)
     IS31FL3733_init(DRIVER_ADDR_1, 0);
+    IS31FL3733_init(DRIVER_ADDR_2, 0);
```
Comment (drashna) [inline quantum/rgb_matrix_drivers.c:37]: This should probably be more like: 
```suggestion
#        defined(DRIVER_ADDR_2) && DRIVER_ADDR_1 != DRIVER_ADDR_2
    IS31FL3733_init(DRIVER_ADDR_2, 0);
#        endif
```

**Comment by drashna** [comment]
For reference (non-comprehensive), the boards that use the 3733 controller are: 
* the DZTech boards, all of them. 
* Exclusive e6_rgb
* hs60 v2 * uses WilbaTech LED
* k_type (ignore, is abomination that uses i2c1 and i2c2), lists different addresses
* kbdfans/kbdmini
* (mass)drop(ped) boards
* miller/gm862
* nebula68 * uses WilbaTech LED
* nk65 * uses WilbaTech LED
# wilba tech boards, or anything that uses their per key rgb (not RGB Matrix feature). 


So the only boards that would need verification/change are the DZTech boards, the e6_rgb, (probably the K_TYPE), the kbdmini, and the gm862.  The drop boards .... not sure

File: quantum/rgb_matrix_drivers.c (hunk 1 above)
Comment (fauxpark) [reply quantum/rgb_matrix_drivers.c:37]: Would prefer to just change the offending boards instead of this hack.

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

**Comment by drashna** [comment]
got a small merge conflict here.

**Comment by Xelus22** [comment]
Closing in favour of #12342

**Pull request closed without merging**